### PR TITLE
Beta: implement square cylinder

### PR DESCRIPTION
**Pull request by PhilPJL**
Implement a 'Square Cylinder' TargetType by patching code using TargetType.Cube.
If TargetParameter2 != 0 then adjust geometry code to use a square cylinder of height TargetParameter2.

TODO: Adjust wire frame drawn showing area of effect after casting the spell - help needed.
TODO: I assume transpilers will reduce the number of patches. Wait until logic fully tested.

### CODE REVIEW

**Comment by ThyWoof** [comment]
I believe all patterns Prefix / Postfix that set Height can be done with less patches if we transpile the call to `UpdateCubePosition_Regular` within `UpdateGeometricShape`.

File: SolastaCommunityExpansion/Settings.cs [hunk 1]
```diff
@@ -169,6 +169,10 @@ public class Settings : UnityModManager.ModSettings
         public bool DismissControlledConjurationsWhenDeliberatelyDropConcentration { get; set; }
         public bool OnlyShowMostPowerfulUpcastConjuredElementalOrFey { get; set; }
         public bool SpikeGrowthDoesNotAffectFlyingCreatures { get; set; }
+        public bool SquareAreaOfEffectSpellsDoNotAffectFlyingCreatures { get; set; } = true;
```
Comment (ChrisPJohn) [inline SolastaCommunityExpansion/Settings.cs:172]: This should probably not be set to true by default.

File: SolastaCommunityExpansion/Spells/HouseSpellTweaks.cs [hunk 2]
```diff
@@ -28,17 +29,101 @@ private static void SpikeGrowthDoesNotAffectFlyingCreatures()
...
+                    .GetDatabase<SpellDefinition>()
+                    .Where(sd => sd.EffectDescription.TargetType == RuleDefinitions.TargetType.Cube))
+                {
+                    // TargetParameter2 is not used by TargetType.Cube but has random values assigned.
+                    // We are going to use it to create a square cylinder with height so set to zero for all spells with TargetType.Cube.
+                    sd.EffectDescription.SetTargetParameter2(0);
+                }
+            }
+
+/*            static void LogAllSpellsWithTargetTypeCube()
```
Comment (ChrisPJohn) [inline SolastaCommunityExpansion/Spells/HouseSpellTweaks.cs:111]: Clean up debug code?

**Review by ChrisPJohn — CHANGES_REQUESTED** [review]
A couple small changes.

For GeometryPatcher I focused on if the code was protected with settings checks and less on the math. I assume I'm unlikely to catch a math error that wasn't caught testing the changes, but also it's late. If you want a thorough math review I can take a look later.

File: SolastaCommunityExpansion/Spells/HouseSpellTweaks.cs (hunk 2 above)
Comment (PhilPJL) [reply SolastaCommunityExpansion/Spells/HouseSpellTweaks.cs]: Removed

**Review by ChrisPJohn — APPROVED** [review]

**Review by ThyWoof — APPROVED** [review]

**Pull request merged**